### DISCUSSION TITLE
Add new CheesHMC sampler by Hoffman et al

### DISCUSSION THREAD

**Discussion by twiecki** [blackjax-devs/blackjax#91] [New features]
http://proceedings.mlr.press/v130/hoffman21a/hoffman21a.pdf

**Comment by rlouf** [comment]
So I had a quick look. It is a new warmup scheme for the most part and here's what would be necessary:
1. a jittered version of HMC, which is trivial to implement here;
2. Dual averaging;
3. Mass matrix adaptation;
4. Their tuning algorithm for the scale of the trajectory length, `T`, which requires the Adam optimizer.

(2) and (3) are part of the Stan warmup and I already have an implementation of Adam in MCX that I can just copy/paste. Will implement once we have a drop in replacement for NUTS + Stan warmup.

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

Implemented as `blackjax.chees_adaptation` (ChEES-HMC warmup). Closing.
